pilot/pkg/bootstrap/server.go

```diff
@@ -742,7 +745,12 @@ func (s *Server) initSecureDiscoveryService(args *PilotArgs) error {
 	}
 	s.SecureGrpcListener = l
 
-	opts := s.grpcServerOptions(args.KeepaliveOptions)
+	interceptors := []grpc.UnaryServerInterceptor{
+		// setup server prometheus monitoring (as final interceptor in chain)
+		prometheus.UnaryServerInterceptor,
+	}
+	log.Infof("using max conn age of %v", args.KeepaliveOptions.MaxServerConnectionAge)
+	opts := xds.GrpcServerOptions(args.KeepaliveOptions, interceptors...)
```

Comment:
Curious why not move the `prometheus.UnaryServerInterceptor` into  GrpcServerOptions

Reply:
We do not need this for proxy